Assets/Scripts/Persona/api/APIClient.cs

```diff
@@ -8,13 +8,18 @@
 
 namespace IO.Persona.MobileAds.Unity
 {
-    public class APIClient
+    public interface IAPIClient
     {
+        Task<string> MakeGetRequestAsync(string url, Dictionary<string, string> queryParams, Dictionary<string, string> headers);
+        Task<string> MakePostRequestAsync(string url, string postData, Dictionary<string, string> queryParams, Dictionary<string, string> headers);
+        Task<Texture2D> DownloadImageFromUrl(string mediaUrl);
```

Comment:
DownloadImageFromUrl and DownloadGifFromUrl - not sure if they are suitable for being part of the API Client interface as they are not really REST API methods ?
Thoughts ?

Reply:
Yeah, agreed. However since the download handler used in UnityWebRequest is different in case of downloading image, gif etc, it appeared more appropriate to place them all in one file which is also making unit test cases easier